src/Bicep.Core/Resources/ResourceTypeReference.cs

```diff
@@ -11,7 +11,7 @@ namespace Bicep.Core.Resources
 {
     public class ResourceTypeReference
     {
-        private static readonly Regex ResourceTypePattern = new Regex(@"^(?<namespace>[a-z0-9][a-z0-9\.]*)(/(?<type>[a-z0-9]+))+@(?<version>(\d{4}-\d{2}-\d{2})(-(preview|alpha|beta|rc|privatepreview))?$)", RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture | RegexOptions.Compiled | RegexOptions.CultureInvariant);
+        private static readonly Regex ResourceTypePattern = new Regex(@"^(?<namespace>[a-z0-9][a-z0-9\.]*)(/(?<type>[a-z0-9\-]+))+@(?<version>(\d{4}-\d{2}-\d{2})(-(preview|alpha|beta|rc|privatepreview))?$)", RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture | RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
>- [](start = 124, length = 1)

Super curious which resource type has a `-` in the type name.

Reply:
Here's the one I ran into: https://github.com/Azure/bicep-types-az/blob/8ab41c41f9b633f57bec3bcd7cfad792dc8b18d3/generated/docs/microsoft.apimanagement/2018-06-01-preview/types.md#microsoftapimanagementserviceapi-version-sets